### ISSUE TITLE
Pseudo dice while training 

### ISSUE THREAD

**Issue by Body123** [MIC-DKFZ/nnUNet#2125]
I want to ask while training in the logs I found as an example :
2024-04-23 15:22:40.711169: Epoch 195
2024-04-23 15:22:40.724555: Current learning rate: 0.00036
2024-04-23 15:23:41.803760: train_loss -0.7526
2024-04-23 15:23:41.832845: val_loss -0.771
2024-04-23 15:23:41.869548: Pseudo dice [0.8399]
2024-04-23 15:23:41.895059: Epoch time: 61.1 s
2024-04-23 15:23:41.911069: Yayy! New best EMA **pseudo Dice: 0.8196** 
but when calculating after the number of epochs end in this fold the number of pseudo dice is less than the above : 
2024-04-23 15:29:06.880623: Mean Validation Dice:  0.6980713938291729
so I want to ask what is this number represents 2024-04-23 15:23:41.869548: Pseudo dice [0.8399] and what this represent :
2024-04-23 15:23:41.911069: Yayy! New best EMA **pseudo Dice: 0.8196**

**Comment by Body123** [comment]
is there any answer?

**Comment by dojoh** [comment]
For performance reasons the pseudo dice is not computed on the whole validation set. The Mean Validation Dice in contrast is computed on the whole dataset, see e.g.:

`progress.png: Shows losses, pseudo dice, learning rate and epoch times ofer the course of the training. At the top is a plot of the training (blue) and validation (red) loss during training. Also shows an approximation of the dice (green) as well as a moving average of it (dotted green line). This approximation is the average Dice score of the foreground classes. It needs to be taken with a big (!) grain of salt because it is computed on randomly drawn patches from the validation data at the end of each epoch, and the aggregation of TP, FP and FN for the Dice computation treats the patches as if they all originate from the same volume ('global Dice'; we do not compute a Dice for each validation case and then average over all cases but pretend that there is only one validation case from which we sample patches). The reason for this is that the 'global Dice' is easy to compute during training and is still quite useful to evaluate whether a model is training at all or not. A proper validation takes way too long to be done each epoch. It is run at the end of the training.`
